src/software/simulated_tests/validation_functions/ball_out_of_bounds_validation.cpp

```diff
@@ -0,0 +1,19 @@
+#include "software/simulated_tests/validation_functions/ball_out_of_bounds_validation.h"
+
+#include "software/new_geom/util/contains.h"
+
+void ballOutOfBounds(std::shared_ptr<World> world_ptr,
+                     ValidationCoroutine::push_type& yield)
+{
+    auto ball_out_of_bounds = [](std::shared_ptr<World> world_ptr) {
+        Point ball_position   = world_ptr->ball().position();
+        Rectangle field_lines = world_ptr->field().fieldLines();
+
+        return contains(field_lines, ball_position);
```

Comment:
This will return `true` if the ball is contained in the field lines, meaning the ball is _not_ out of bounds